durable-storage/benches/avl_tree.rs

```diff
@@ -0,0 +1,177 @@
+// SPDX-FileCopyrightText: 2025 TriliTech <contact@trili.tech>
+// SPDX-License-Identifier: MIT
+
+use std::collections::BTreeMap;
+use std::time::Duration;
+
+use bytes::Bytes;
+use criterion::Criterion;
+use criterion::criterion_group;
+use criterion::criterion_main;
+use octez_riscv_durable_storage::merkle_layer::KEY_MAX_SIZE;
+use octez_riscv_durable_storage::merkle_layer::Key;
+use octez_riscv_durable_storage::merkle_layer::tree::Avl;
+use rand::prelude::*;
+use rand::random_range;
+
+#[derive(Debug, Clone)]
+pub enum Operation {
+    Get(Key),
+    Upsert(Key, Bytes),
+    Delete(Key),
+}
+
+fn generate_byte_vector(rng: &mut impl Rng, length: usize) -> Vec<u8> {
+    let mut ret = vec![0u8; length];
+    rng.fill(ret.as_mut_slice());
+    ret
+}
+
+fn generate_keys(rng: &mut impl Rng, length: usize) -> Vec<Key> {
+    let mut ret: Vec<Key> = Vec::with_capacity(length);
+    for _ in 0..length {
+        let key_length = rng.random_range(1..KEY_MAX_SIZE);
+        ret.push(
+            Key::new(&generate_byte_vector(rng, key_length))
+                .expect("The byte vector is the right size"),
+        );
+    }
+    ret
+}
+
+fn get_random_data(rng: &mut impl Rng) -> Bytes {
+    let length: usize = rng.random_range(1..20);
+    let bytes = generate_byte_vector(rng, length);
+    Bytes::from(bytes)
+}
+
+fn get_operations_batch(rng: &mut impl Rng, keys: &[Key], batch_size: usize) -> Vec<Operation> {
+    (0..batch_size)
+        .map(|_| {
+            let key: Key = keys
+                .choose(rng)
+                .expect("The keys array is not empty")
+                .clone();
+            match random_range(0..3) {
```

Comment:
```suggestion
            match rng.random_range(0..3) {
```
Consistency + `rand::random_range` import no longer needed